tests/compatibility_trait.php

```diff
@@ -0,0 +1,73 @@
+<?php
+// This file is part of Moodle - https://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
```

Comment:
Can we remove this block, as its duplcated against the trait phpDoc? Not sure if the CI tools are smart enough to handle traits correctly as they do for classes

Reply:
I believe trait docblocks are required because they aren't handled as well as classes. I could give this another try locally.